build.sh

```diff
@@ -3,132 +3,134 @@
 set -euo pipefail
 
 clean_modules() {
-  depth=1
+  local depth=1 to_del
   while true ; do
-    toDel=$( find . -maxdepth $depth -type d -name node_modules )
-    echo $toDel
-    if [[ "$toDel" == '' ]] ; then return 0 ; fi
-    rm -rf $toDel
+    to_del=$( find . -maxdepth $depth -type d -name node_modules )
+    echo "$to_del"
+    if [[ "$to_del" == '' ]] ; then
+      return 0
+    fi
+    rm -rf "$to_del"
     depth=$(( depth + 1 ))
   done
 }
 
-export ENV=$1
-S3_FOLDER_NAME=${GITHUB_RUN_ID:-}
-
-echo "Configuring aws cli."
-mkdir -p ~/.aws
-
-cat > ~/.aws/config <<eoco
+function main() {
+  if [[ ! -d ~/.aws ]] ; then
+  echo "Configuring aws cli."
+  mkdir -p ~/.aws
+  cat > ~/.aws/config <<eoco
 [herokles]
 region = $HEROKLES_AWS_REGION
 eoco
 
-cat > ~/.aws/credentials <<eocre
+  cat > ~/.aws/credentials <<eocre
 [herokles]
 aws_access_key_id = $HEROKLES_AWS_ACCESS_KEY_ID
 aws_secret_access_key = $HEROKLES_AWS_SECRET_ACCESS_KEY
 region = $HEROKLES_AWS_REGION
 eocre
-
-echo "Getting environment variables."
-JSON=$( mktemp )
-JSON_TEMPLATE=$( mktemp )
-JSON_FULL=$( mktemp )
-aws --profile herokles ssm get-parameters --name /${PROJECT}/${ENV} > $JSON_FULL
-
-if [[ $ENV == pr-${PR_NUM:-''} ]] ; then
-  aws --profile herokles ssm get-parameters --name /${PROJECT}/prs > $JSON_TEMPLATE # get all template envs
-  if [[ ! -z $( jq -r '.InvalidParameters | .[]' $JSON_TEMPLATE ) ]] ; then
-    echo "Template PR variables /${PROJECT}/prs not found."
-    exit 1
-  fi
-  if [[ ! -z $( jq -r '.InvalidParameters | .[]' $JSON_FULL ) ]] ; then # check if it's a new PR
-    echo "New PR - setting to copy from /${PROJECT}/prs."
-    echo '{ "Parameters": [ { "Value": { } } ] }' > $JSON_FULL
   fi
-  # find new envs in /PROJECT/prs template, fill them in and push
-  # $update file works a) for retrieving only the env var values from a rich JSON and b) as a condition checker for when to update pr-<num> parameter
-  update=$( mktemp )
-  jq -r '.Parameters | .[] | .Value' $JSON_FULL > $update
-  cat $update > $JSON
-  jq -r '.Parameters | .[] | .Value' $JSON_TEMPLATE > $update
-  cat $update > $JSON_TEMPLATE
-  echo > $update
-  for key in $( jq -r 'keys[]' $JSON_TEMPLATE ) ; do
-    if [[ $( jq -r .$key $JSON ) == null ]] ; then
-      cat $JSON | jq ".${key}=\"$( jq -r .$key $JSON_TEMPLATE )\"" > $update
-      cat $update > $JSON
+
+  echo "Getting environment variables."
+  local JSON=$( mktemp )
+  local JSON_TEMPLATE=$( mktemp )
+  local JSON_FULL=$( mktemp )
```

Comment:
We are not doing google style to the letter. However one thing I would like to see us adopted is to use capital lettered var names only for
1. vars coming from environment
2. vars intended to be exported to environment
3. readonly vars declared at the very beginning of script.

Other vars (declared and used inside the script) should be lower case.